sdc/hiframes/boxing.py

```diff
@@ -100,9 +100,19 @@ def unbox_dataframe(typ, val, c):
 
     column_tup = c.context.make_tuple(
         c.builder, types.UniTuple(string_type, n_cols), column_strs)
-    zero = c.context.get_constant(types.int8, 0)
-    unboxed_tup = c.context.make_tuple(
-        c.builder, types.UniTuple(types.int8, n_cols + 1), [zero] * (n_cols + 1))
+
+    # this unboxes all DF columns so that no column unboxing occurs later
+    for col_ind in range(n_cols):
+        series_obj = c.pyapi.object_getattr_string(val, typ.columns[col_ind])
+        arr_obj = c.pyapi.object_getattr_string(series_obj, "values")
+        ty_series = typ.data[col_ind]
+        if isinstance(ty_series, types.Array):
+            native_val = unbox_array(typ.data[col_ind], arr_obj, c)
+        elif ty_series == string_array_type:
+            native_val = unbox_str_series(string_array_type, series_obj, c)
+
+        dataframe.data = c.builder.insert_value(
```

Comment:
I believe we can set data once, not in every iteration

Reply:
This code works like this:
```
data1 = {undef,undef,undef,undef}
data2 = insert(data1, val1, 0) # {val1, undef, undef, undef}
data3 = insert(data2, val2, 1) # {val1, val2, undef, undef}
data4 = insert(data3, val3, 2) # {val1, val2, val3, undef}
...etc...
```

Reply:
Can't it be something like:
```python
    data_tup = c.context.make_tuple(
        c.builder, types.Tuple(types), values)
```
?

Not sure how to get types though

Reply:
meh, not worth it